js/bitmarket.js

```diff
@@ -185,6 +180,96 @@ module.exports = class bitmarket extends Exchange {
         });
     }
 
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetJsonInternalAllTicker (this.extend ({}, params));
+        const ids = Object.keys (response);
+        const result = [];
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            const item = response[id];
+            const baseId = id.substring (0, 3);
+            const quoteId = id.substring (3);
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            result.push ({
+                'id': id,
+                'symbol': base + '/' + quote,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': undefined,
+                'info': item,
+            });
+        }
+        return result;
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.privatePostWithdrawals (this.extend (request, params));
+        const items = response['data']['results'];
+        return this.parseTransactions (items, undefined, since, limit);
+    }
+
+    parseTransaction (item, currency = undefined) {
+        //   { id: 240565,
+        //     transaction_id:
+        //     '78cbf0405f07a578164644aa67f5c6a08197574bc100a50aaee40ef2e11dc2d7',
+        //         received_in: '1EdAqY4cqHoJGAgNfUFER7yZpg1Jc9DUa3',
+        //     currency: 'BTC',
+        //     amount: 0.49926113,
+        //     time: 1518353534,
+        //     commission: 0.0008,
+        //     withdraw_type: 'Cryptocurrency' }
+        const ccyCode = this.safeString (item, 'currency');
+        const time = this.safeInteger (item, 'time');
+        const timestamp = time * 1000;
+        const ccyId = this.currencyId (ccyCode);
```

Comment:
`currencyId` gets a unified currency code and returns an currency id...
In this case we need the opposite – to convert from a currency id, to a common currency code, which is usually done in the following way:

```Javascript
let code = undefined;
const currencyId = this.safeString (item, 'currency');
if (currencyId !== undefined) {
    if (currencyId in this.currencies_by_id) {
        code = this.currencies_by_id[currencyId]['code'];
    } else {
        code = this.commonCurrencyCode (currencyId);
    }
}
```

Reply:
It just so happens that for this exchange codes are equal to ids, so the bug does not show up, but still does live here (may break it in the future).

Reply:
Opps, thanks.